File: flying-ui/src/api/node.js
import request from '@/router/axios';
export function getNodeAll() {
    return request({
        url: '/node/getNodeList',
        method: 'get',
    })
}
export function getAppNode(appId) {
    return request({
        url: '/node/getAppNode/'+appId,
        method: 'get',

    })
}
export function getSyncNode(query) {
    return request({
        url: '/node/getSyncNode',
        method: 'get',
        params: query

    })
}
export function getNodeList(query) {
    return request({
        url: '/node/getNodeList',
        method: 'get',
        params: query
    })
}
export function deleteNode(obj) {
    return request({
        url: '/node/deleteNode',
        method: 'delete',
        data: obj
    })
}
export function createNode(obj) {
    return request({
        url: '/node/createNode',
        method: 'post',
        data: obj
    })
}
export function findNode(id) {
    return request({
        url: '/node/findNode/'+id,
        method: 'get',

    })
}
export function updateNode(obj) {
    return request({
        url: '/node/updateNode',
        method: 'put',
        data: obj

    })
}
export function ping(id) {
    return request({
        url: '/node/ping/'+id,
        method: 'get',

    })
}

File: flying-admin/router/node.go
package router

import (
	"flying-admin/api"
	"github.com/gin-gonic/gin"
)

func InitNodeRouter(Router *gin.RouterGroup) {
	NodeRouter := Router.Group("node")
	{
		NodeRouter.POST("createNode", api.CreateNode)             // 新建Node
		NodeRouter.DELETE("deleteNode", api.DeleteNode)           // 删除Node
		NodeRouter.DELETE("deleteNodeByIds", api.DeleteNodeByIds) // 批量删除Node
		NodeRouter.PUT("updateNode", api.UpdateNode)              // 更新Node
		NodeRouter.GET("findNode/:id", api.FindNode)              // 根据ID获取Node
		NodeRouter.GET("getNodeList", api.GetNodeList)            // 获取Node列表
		NodeRouter.GET("getNodeAll", api.GetNodeAll)              // 获取所有的环境
		NodeRouter.GET("ping/:id", api.Ping)                      // 测试是否连接
		NodeRouter.GET("getAppNode/:appId", api.GetAppNode)       // 获取当前app的环境
		NodeRouter.GET("getSyncNode", api.GetSyncNode)            // 获取要进行同步的namespace
	}
}
